components/menu/submenu-inline-child.component.ts

```diff
@@ -40,63 +40,50 @@ import { NzMenuModeType } from './menu.types';
   },
   imports: [NgTemplateOutlet]
 })
-export class NzSubmenuInlineChildComponent implements OnDestroy, OnInit, OnChanges {
+export class NzSubmenuInlineChildComponent implements OnInit, OnChanges {
+  private readonly elementRef = inject(ElementRef);
+  private readonly renderer = inject(Renderer2);
+  private readonly directionality = inject(Directionality);
+  private readonly destroyRef = inject(DestroyRef);
+
   @Input() templateOutlet: TemplateRef<NzSafeAny> | null = null;
   @Input() menuClass: string = '';
   @Input() mode: NzMenuModeType = 'vertical';
   @Input() nzOpen = false;
   listOfCacheClassName: string[] = [];
   expandState = 'collapsed';
   dir: Direction = 'ltr';
-  private destroy$ = new Subject<void>();
-
-  constructor(
-    private elementRef: ElementRef,
-    private renderer: Renderer2,
-    private directionality: Directionality
-  ) {}
 
   calcMotionState(): void {
-    if (this.nzOpen) {
-      this.expandState = 'expanded';
-    } else {
-      this.expandState = 'collapsed';
-    }
+    this.expandState = this.nzOpen ? 'expanded' : 'collapsed';
   }
+
   ngOnInit(): void {
     this.calcMotionState();
 
     this.dir = this.directionality.value;
-    this.directionality.change?.pipe(takeUntil(this.destroy$)).subscribe((direction: Direction) => {
+    this.directionality.change?.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(direction => {
       this.dir = direction;
     });
   }
+
   ngOnChanges(changes: SimpleChanges): void {
     const { mode, nzOpen, menuClass } = changes;
     if (mode || nzOpen) {
       this.calcMotionState();
     }
     if (menuClass) {
       if (this.listOfCacheClassName.length) {
-        this.listOfCacheClassName
-          .filter(item => !!item)
-          .forEach(className => {
-            this.renderer.removeClass(this.elementRef.nativeElement, className);
-          });
+        this.listOfCacheClassName.filter(Boolean).forEach(className => {
+          this.renderer.removeClass(this.elementRef.nativeElement, className);
+        });
```

Comment:
Reduce the number of traversals:
```suggestion
        this.listOfCacheClassName.forEach(className => {
          if (className) {
            this.renderer.removeClass(this.elementRef.nativeElement, className);
          }
        });
```